includes/Core/Authentication/Authentication.php

```diff
@@ -547,52 +598,67 @@ private function inline_js_admin_data( $data ) {
-			$data['clientSecret'] = str_repeat( '•', strlen( $client_data['oauth2_client_secret'] ) );
-		} else {
+			// TODO: Remove once related JS functionality is removed. For now, still set these as false-y.
+			$data['clientID']     = '';
 			$data['clientSecret'] = '';
-		}
-
-		if ( current_user_can( Permissions::MANAGE_OPTIONS ) && $apikey ) {
-			$data['apikey'] = $apikey;
+			$data['apikey']       = false;
+			$data['projectId']    = false;
+			$data['projectUrl']   = false;
 		} else {
-			$data['apikey'] = false;
-		}
+			// TODO: Remove once related JS functionality is removed.
+			$client_data = $this->credentials->get();
+			$apikey      = $this->get_api_key_client()->get_api_key();
+			$gcp_project = $this->gcp_project->get();
+
+			if ( current_user_can( Permissions::MANAGE_OPTIONS ) && isset( $client_data['oauth2_client_id'] ) ) {
+				$data['clientID'] = $client_data['oauth2_client_id'];
+			} else {
+				$data['clientID'] = '';
+			}
+			if ( current_user_can( Permissions::MANAGE_OPTIONS ) && isset( $client_data['oauth2_client_secret'] ) ) {
+				$data['clientSecret'] = str_repeat( '•', strlen( $client_data['oauth2_client_secret'] ) );
+			} else {
+				$data['clientSecret'] = '';
+			}
 
-		// Make GCP project information available only to the creator.
-		if ( ! empty( $gcp_project['id'] ) && (int) get_current_user_id() === $gcp_project['wp_owner_id'] ) {
-			$data['projectId']  = $gcp_project['id'];
-			$data['projectUrl'] = add_query_arg( 'project', $gcp_project['id'], 'https://console.cloud.google.com/apis/credentials' );
-		} else {
-			$data['projectId']  = false;
-			$data['projectUrl'] = false;
-		}
+			if ( current_user_can( Permissions::MANAGE_OPTIONS ) && $apikey ) {
+				$data['apikey'] = $apikey;
+			} else {
+				$data['apikey'] = false;
+			}
 
-		$data['connectUrl']    = esc_url_raw( $this->get_connect_url() );
-		$data['disconnectUrl'] = esc_url_raw( $this->get_disconnect_url() );
+			// Make GCP project information available only to the creator.
+			if ( ! empty( $gcp_project['id'] ) && (int) get_current_user_id() === $gcp_project['wp_owner_id'] ) {
+				$data['projectId']  = $gcp_project['id'];
+				$data['projectUrl'] = add_query_arg( 'project', $gcp_project['id'], 'https://console.cloud.google.com/apis/credentials' );
+			} else {
+				$data['projectId']  = false;
+				$data['projectUrl'] = false;
+			}
 
-		$external_sitename = html_entity_decode( get_bloginfo( 'name' ), ENT_QUOTES );
-		$external_sitename = str_replace( '&', 'and', $external_sitename );
-		$external_sitename = trim( preg_replace( '/([^A-Za-z0-9 ]+|google)/i', '', $external_sitename ) );
-		if ( strlen( $external_sitename ) < 4 ) {
-			$external_sitename .= ' Site Kit';
+			$external_sitename = html_entity_decode( get_bloginfo( 'name' ), ENT_QUOTES );
+			$external_sitename = str_replace( '&', 'and', $external_sitename );
+			$external_sitename = trim( preg_replace( '/([^A-Za-z0-9 ]+|google)/i', '', $external_sitename ) );
```

Comment:
This was a very special case for Henhouse. It's actually just existing code being moved here, and it now actually can be removed since #560 has already been merged (no more Henhouse).

Reply:
Actually, this needs to remain here depending on #511. Let's remove depending on which of these two PRs get merged first.